spec/tbd_tests_spec.rb

```diff
@@ -1566,3 +1566,414 @@
+        "unit": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Unit"
+          },
+          "uniqueItems": true
+        },
+        "logs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Log"
+          }
+        }
+      },
+      "additionalProperties": false,
+      "definitions": {
+        "PSI": {
+          "description": "Set of PSI-values (in W/K per m) for thermal bridges",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "Unique PSI set identifier",
+              "type": "string"
+            },
+            "rimjoist": {
+              "title": "Floor/wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "parapet": {
+              "title": "Roof/wall or exposed-floor/wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "fenestration": {
+              "title": "Window, door, skylight perimeter PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "concave": {
+              "title": "Wall corner [0°,135°) PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "convex": {
+              "title": "Wall corner (225°,360°] PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "balcony": {
+              "title": "Floor/balcony edge PSI ",
+              "type": "number",
+              "minimum": 0
+            },
+            "party": {
+              "title": "Party wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "grade": {
+              "title": "Floor/foundation edge PSI",
+              "type": "number",
+              "minimum": 0
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id"
+          ],
+          "minProperties": 2
+        },
+        "KHI": {
+          "description": "KHI-value (in W/K) for a point thermal bridge",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "Unique KHI identifier",
+              "type": "string"
+            },
+            "point": {
+              "title": "Point KHI-value",
+              "type": "number",
+              "minimum": 0
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id",
+            "point"
+          ]
+        },
+        "Edge": {
+          "description": "Surface(s) edge as thermal bridge",
+          "type": "object",
+          "properties": {
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            },
+            "type": {
+              "title": "PSI-set type, e.g. 'parapet'",
+              "type": "string"
```

Comment:
Do you want to enumerate the allowable types?

Reply:
https://json-schema.org/understanding-json-schema/reference/generic.html#id4

Reply:
Good point. I'll use the [enum](https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values) keyword. I was planning to add a post-schema validation control to flag unrecognized types here, i.e. "doorjamb" - enum is far better. Thanks.